patches/sitecustomize.py

```diff
@@ -0,0 +1,23 @@
+# Monkey patches BigQuery client creation to use data proxy
+import os
+
+kaggle_proxy_data = os.getenv("KAGGLE_PROXY_DATA")
+if kaggle_proxy_data:
+    from google.auth import credentials
+    from google.cloud import bigquery
+    from google.cloud.bigquery._http import Connection
+    from google.cloud import _http
+    from google.cloud.bigquery import __version__
+
+    Connection.API_BASE_URL = os.getenv("KAGGLE_PROXY_URL")
+
+    _CLIENT_INFO = _http.CLIENT_INFO_TEMPLATE.format(__version__)
+    Connection._EXTRA_HEADERS = {
+        _http.CLIENT_INFO_HEADER: _CLIENT_INFO,
+        'X-KAGGLE-PROXY-DATA': kaggle_proxy_data
+    }
```

Comment:
Would it be less fragile to add to the existing dictionary (assuming there is one) rather than redefining it?

```
Connection._EXTRA_HEADERS['X-KAGGLE-PROXY-DATA'] = kaggle_proxy_data
```

There probably is a reason this isn't done, but I figured I'd ask.

Reply:
Just took this from the existing code, but don't see why that wouldn't work. Will try.